W-11086312 and W-11174370: Added cases to Common Errors (#2225)
mulesoft/docs-mule-runtime: modules/ROOT/pages/mule-upgrade-tool.adoc

Verify that the downloaded Mule distribution is in a later version than your current Mule instance. If you continue to receive this error message during the upgrade, it means that the current Mule instance is already updated or running the latest available version.

=== Missing reading permissions

----
./upgrade-tool upgrade -n /tmp/mule-enterprise-standalone-4.4.0-20211104

Mule Runtime Upgrade Tool
─────────────────────────

INFO: Initiating upgrade process... 
INFO: Upgrading distribution at path [/opt/mule-enterprise-standalone-4.4.0-20211026] with version: [4.4.0-20211026] to distribution at path [/tmp/mule-enterprise-standalone-4.4.0-20211104] with version [4.4.0-20211104] ...
ERROR: Error validating required reading permissions under Mule distribution [/tmp/mule-enterprise-standalone-4.4.0-20211104]
java.nio.file.AccessDeniedException: /tmp/mule-enterprise-standalone-4.4.0-20211104/lib/mule/mule-module-cache-ee-4.4.0-20211104.jar
[...]
----

==== Cause

> Reading permissions in the new Mule distribution are needed, and the file referenced in the stack trace is not readable by the user performing the upgrade.

Reviewer: ```suggestion
Read permissions in the new Mule distribution are required but not set for the user who is performing the upgrade, so that user cannot read the file referenced in the stack trace.
```

==== Solution

> The upgrade tool helps on identifying which files are not readable, yet, assistance from the system administrator could be needed to correct permissions.

Reviewer: ```suggestion
Obtain read permission for files that the upgrade identifies as unreadable. Contact your system administrator for assistance, if necessary.
```

> The upgrade tool helps on identifying which files are not writtable, yet, assistance from the system administrator could be needed to correct permissions.

Author: ```suggestion
The upgrade tool helps on identifying which files are not readable, yet, assistance from the system administrator could be needed to correct permissions.
```

=== Missing writing permissions

----
./upgrade-tool upgrade -n /tmp/mule-enterprise-standalone-4.4.0-20211104

Mule Runtime Upgrade Tool
─────────────────────────

INFO: Initiating upgrade process... 
INFO: Upgrading distribution at path [/opt/mule-enterprise-standalone-4.4.0-20211026] with version: [4.4.0-20211026] to distribution at path [/tmp/mule-enterprise-standalone-4.4.0-20211104] with version [4.4.0-20211104] ...
ERROR: Error validating required writing permissions under Mule distribution [/opt/mule-enterprise-standalone-4.4.0-20211026]
java.nio.file.AccessDeniedException: /opt/mule-enterprise-standalone-4.4.0-20211026/lib/boot/libwrapper-linux-ppcle-64.so
[...]
----

==== Cause

> Writting permissions in the old Mule distribution are needed, and the file referenced in the stack trace is not writtable by the user performing the upgrade.

Reviewer: ```suggestion
Write permissions in the old Mule distribution are required but not set for the user who is performing the upgrade, so that user cannot write to the file referenced in the stack trace.
```

==== Solution

> The upgrade tool helps on identifying which files are not writtable, yet, assistance from the system administrator could be needed to correct permissions.

Reviewer: ```suggestion
Obtain write permission to files that the upgrade tool identifies. Contact your system administrator for assistance, if necessary.
```

=== Mule Runtime should be stopped

----
./upgrade-tool upgrade -n /tmp/mule-enterprise-standalone-4.4.0-20211104

Mule Runtime Upgrade Tool
─────────────────────────

INFO: Initiating upgrade process... 
INFO: Upgrading distribution at path [/opt/mule-enterprise-standalone-4.4.0-20211026] with version: [4.4.0-20211026] to distribution at path [/tmp/mule-enterprise-standalone-4.4.0-20211104] with version [4.4.0-20211104] ...
ERROR: Mule Runtime should be stopped.
ERROR: Not all preconditions are met to perform the upgrade. Review the log for details.
Not all preconditions are met to perform the upgrade. Review the log for details.
----

==== Cause

> The upgrade tool relies on the information returned by the Mule Runtime to determine if it is running or not. For the purpose of the upgrade tool, it should not be running.

Reviewer: ```suggestion
The upgrade tool detected that Mule is running. 
```

==== Solution

> The Mule Runtime should be stopped before starting the upgrade process. To check the current status use the command `${MULE_HOME}/bin/mule status`.

Reviewer: ```suggestion
Stop Mule before starting the upgrade process. To check the current status, use the command `${MULE_HOME}/bin/mule status`.
```
